src/status_im2/contexts/communities/menus/community_options/view.cljs

```diff
@@ -0,0 +1,177 @@
+
+(defn community-notification-settings
+  [id]
+  {:icon                :i/notifications
+   :accessibility-label :community-notification-settings
+   :label               (i18n/label :t/community-notification-settings)
+   :on-press            #(js/alert (str "implement action" id))
+   :right-icon          :i/chevron-right})
+
+(defn invite-contacts
+  [id]
+  {:icon                :i/add-user
+   :accessibility-label :invite-people-from-contacts
+   :label               (i18n/label :t/invite-people-from-contacts)
+   :on-press            #(hide-sheet-and-dispatch [:communities/invite-people-pressed id])})
+
+(defn show-qr
+  [id]
+  {:icon                :i/qr-code
+   :accessibility-label :show-qr
+   :on-press            #(js/alert (str "implement action" id))
+   :label               (i18n/label :t/show-qr)})
+
+(defn share-community
+  [id]
+  {:icon                :i/share
+   :accessibility-label :share-community
+   :label               (i18n/label :t/share-community)
+   :on-press            #(hide-sheet-and-dispatch [:communities/share-community-pressed id])})
+
+(defn leave-community
+  [id]
+  {:icon                :i/log-out
+   :label               (i18n/label :t/leave-community)
+   :accessibility-label :leave-community
+
+   :danger?             true
+   :on-press            #(rf/dispatch [:bottom-sheet/show-sheet
+                                       {:content        (constantly [leave-menu/leave-sheet id])
+                                        :content-height 400}])})
+
+(defn cancel-request-to-join
+  [id]
+  {:icon                :i/block
+   :label               (i18n/label :t/cancel-request-to-join)
+   :accessibility-label :cancel-request-to-join
+   :danger?             true
+   :on-press            #(js/alert (str "implement action" id))})
+
+(defn edit-community
+  [id]
+  {:icon                :i/edit
+   :label               (i18n/label :t/edit-community)
+   :accessibility-label :edit-community
+   :on-press            #(js/alert (str "implement action" id))})
+
+(defn not-joined-options
+  [id token-gated?]
+  [[(when-not token-gated? (view-members id))
+    (when-not token-gated? (view-rules id))
+    (invite-contacts id)
+    (when token-gated? (view-token-gating id))
+    (show-qr id)
+    (share-community id)]])
+
+(defn join-request-sent-options
+  [id token-gated?]
+  [(conj (first (not-joined-options id token-gated?)) (cancel-request-to-join id))])
+
+(defn join-request-denied
+  [id token-gated?]
+  (not-joined-options id token-gated?))
+
+
+(defn banned-options
+  [id token-gated?]
+  (not-joined-options id token-gated?))
+
+(defn joined-options
+  [id token-gated? muted?]
+  [[(view-members id)
+    (view-rules id)
+    (when token-gated? (view-token-gating id))
+    (mark-as-read id)
+    (mute-community id muted?)
+    (community-notification-settings id)
+    (invite-contacts id)
+    (show-qr id)
+    (share-community id)]
+   [(leave-community id)]])
+
+(defn owner-options
```

Comment:
yeah that sounds fair, we can work to get this in and create a follow up task to communicate with the Desktop team so we can get this logic from their side. CC @cammellos - who can I can discuss with about topics like this?

Reply:
Thanks @cammellos for sharing the current strategy.

The conversation was private, but it was related to the Activity Center and if `status-go` should be responsible or not for telling if a notification requires a call to action or not. Currently the desktop and mobile decide on the front because it's much more convenient. That's why I made the comment in this PR, because it looks like the kind of logic that could be shared between clients.

But I think we haven't necessarily agreed on what a capability really is in the first place, so maybe this word is bringing more confusion to the discussion.

I was under the impression we wanted to follow a more "heavy server, light client" kind of strategy, so that we reduce the chances of clients drifting (when they use the same status-go version of course).

Maybe I just got things wrong, but it might be beneficial to bring this topic to some people in the desktop team as well. We might all have different perspectives on what should reside on the backend vs frontend.

Reply:
Sure,
feel free to set up a conversation with the relevant people, I'll be happy to join!

Reply:
add me too @ilmotta 👍